lib/p2p/Pool.ts

```diff
@@ -412,10 +431,11 @@ class Pool extends EventEmitter {
         await this.nodes.createNode({
           addresses,
           nodePubKey: peer.nodePubKey!,
+          lastAddress: peer.address,
         });
       } else {
         // the node is known, update its listening addresses
-        await this.nodes.updateAddresses(peer.nodePubKey!, addresses);
+        await this.nodes.updateAddresses(peer.nodePubKey!, addresses, peer.address);
```

Comment:
We only want to set a value for `lastAddress` if the connection is outbound (similar to the check above for updating `lastConnected` on `addresses`). Same for the new node case below.

Reply:
Oh right, I guess I overlooked it, since I need to set it for only outbound connections, I have  to make it optional.